tests/test_datasets/test_dataset.py

```diff
@@ -956,18 +957,35 @@ def test_train_test_split_pass_regressors_to_output(df_and_regressors):
     df, df_exog, known_future = df_and_regressors
     ts = TSDataset(df=df, df_exog=df_exog, freq="D", known_future=known_future)
     train, test = ts.train_test_split(test_size=5)
+    assert set(train.regressors).issubset(set(train.features))
+    assert set(test.regressors).issubset(set(test.features))
+    assert train.regressors == ts.regressors
+    assert test.regressors == ts.regressors
+
+
+def test_train_test_split_pass_transform_regressors_to_output(df_and_regressors):
+    df, df_exog, known_future = df_and_regressors
+    ts = TSDataset(df=df, df_exog=df_exog, freq="D", known_future=known_future)
+    ts.fit_transform(transforms=[LagTransform(in_column="target", lags=[1, 2, 3])])
+    train, test = ts.train_test_split(test_size=5)
+    assert set(train.regressors).issubset(set(train.features))
+    assert set(test.regressors).issubset(set(test.features))
     assert train.regressors == ts.regressors
     assert test.regressors == ts.regressors
 
 
 def test_train_test_split_pass_target_components_to_output(ts_with_target_components):
     train, test = ts_with_target_components.train_test_split(test_size=5)
+    assert set(train.target_components_names).issubset(set(train.features))
+    assert set(test.target_components_names).issubset(set(test.features))
     assert sorted(train.target_components_names) == sorted(ts_with_target_components.target_components_names)
     assert sorted(test.target_components_names) == sorted(ts_with_target_components.target_components_names)
 
 
 def test_train_test_split_pass_prediction_intervals_to_output(ts_with_prediction_intervals):
     train, test = ts_with_prediction_intervals.train_test_split(test_size=5)
+    assert set(train.prediction_intervals_names).issubset(set(train.features))
```

Comment:
These conditions failed on previous version of the code.

Reply:
Similar to target components